sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -98,8 +98,8 @@ object TypeUtils {
     case _ => false
   }
 
-  def failWithIntervalType(dataType: DataType): Unit = {
-    invokeOnceForInterval(dataType, forbidAnsiIntervals = true) {
+  def failWithIntervalType(dataType: DataType, forbidAnsiIntervals: Boolean = true): Unit = {
```

Comment:
BTW, where do we disallow ansi interval now?

Reply:
We disallow in ALTER TABLE .. ADD COLUMNS and REPLACE COLUMNS. I plan to add unified tests (also need to check v1) and enable ANSI intervals in the commands.